server/xpub-controller/manuscript.js

```diff
@@ -1,17 +1,192 @@
 const logger = require('@pubsweet/logger')
 const ManuscriptModel = require('@elifesciences/xpub-model').Manuscript
-const UserModel = require('@elifesciences/xpub-model').User
+const FileModel = require('@elifesciences/xpub-model').File
+
 const Notification = require('./notification')
 
 class Manuscript {
-  constructor(config, user) {
+  constructor(config, user, storage, scienceBeamApi, pubsubManager) {
     this.user = user
     this.config = config
+    this.storage = storage
+    this.scienceBeamApi = scienceBeamApi
+    this.pubsubManager = pubsubManager
+  }
+
+  async upload(manuscriptId, file, fileSize, userId) {
```

Comment:
So currently the client requires the `ws` message to be `${ON_UPLOAD_PROGRESS}.${userId}` as seen on line 55. I don't really like having to pass `userId` down as a param as we already have `this.user` available BUT these are two differently derived values as seen in https://github.com/elifesciences/elife-xpub/blob/eb62007acfced2a188bfa72466316d66ce5b20bf/server/xpub-model/entities/user/index.js#L48

I don't think the client has access to this so it's not possible to change these around 👎 any idea's on how we might be able to escape the need to pass this?

Reply:
Maybe there should be a check before calling this that `manuscript.user == userId` ?

Reply:
It looks like we should be able to pass the entire `user` when instantiating the manuscript, and then call `User.getUuidForProfile` within the controller. It does mean that the resolver is basically empty, and delegating everything to the controller, which is defeating the purpose of having this separation...

The only place we need to look at the full user, is this section: 
```
    // ensure user can view manuscript
    let manuscript = await ManuscriptModel.find(manuscriptId, this.user)
```

We can leave this in the resolver, which means we have resolvers authorizing the call, and the controller doing the processing.

@diversemix what do you think?

Reply:
Let's leave architectural conversations till next week. I would argue that a `find` function is not the way we should be checking permissions.

Reply:
Semra and I went over this and we came to the same conclusion.